src/sbd-cluster.c

```diff
@@ -168,10 +173,19 @@ static void sbd_cmap_notify_fn(
 }
 
 static gboolean
-cmap_dispatch_callback (gpointer user_data)
+cmap_dispatch_callback (gint cmap_fd,
+                        GIOCondition condition,
+                        gpointer user_data)
 {
+    /* CMAP connection lost */
+    if (condition & G_IO_HUP) {
+        cl_log(LOG_WARNING, "CMAP service connection lost\n");
+        cmap_destroy();
+        /* remove the source from the main loop */
+        return G_SOURCE_REMOVE;
+    }
     cmap_dispatch(cmap_handle, CS_DISPATCH_ALL);
-    return TRUE;
+    return G_SOURCE_CONTINUE;
```

Comment:
G_SOURCE_REMOVE and G_SOURCE_CONTINUE were added in glib 2.32 -- does anyone know what minimum version sdb supports?

Reply:
For rhel the current sources are compatible with rhel-6.x (2.28 afaik). But shared disk isn't enabled with that so enabling this code branch wouldn't make sense (might even be automatically disabled with corosync 1.4.x).
Would probably make sense to define G_SOURCE_REMOVE and G_SOURCE_CONTINUE if not already defined here - as already done with G_SOURCE_FUNC.